tests/tests.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright 2016 Gomez Guillaume
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fs::{File, OpenOptions, create_dir, remove_file};
+use std::io::{Read, Write};
+use std::path::Path;
+
+extern crate stripper_lib;
+
+const TEST_FILE_DIR : &'static str = "tests/files";
+
+const BASIC : &'static str = r#"/// struct Foo comment
+struct Foo {
+    /// Foo comment
+    /// fn some_func(a: u32,
+    ///              b: u32) {}
+    A: u32,
+}
+
+mod Bar {
+    test! {
+        /// struct inside macro
+        struct SuperFoo;
+        sub_test! {
+            /// and another one!
+            struct FooFoo {
```

Comment:
It's not realistic to have this kind of nesting. This and the `mod Bar§macro test!§struct SuperFoo` feature they're lines of code that are going to sit there unused, getting in the way of features we need and bug fixes.